go/processor/floria/precompiled_test.go

```diff
@@ -80,3 +81,347 @@ func TestPrecompiled_AddressesAreHandledCorrectly(t *testing.T) {
+				state.EXPECT().SetCode(tosca.Address{0x01}, tosca.Code(make([]byte, codeSize)))
+			}
+
+			gas, err := stateSwapCode(state, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != 0 {
+				t.Errorf("unexpected gas, want %v, got %v", 0, gas)
+			}
+		})
+	}
+}
+
+func TestPrecompiled_StateSetStorage(t *testing.T) {
+	tests := map[string]struct {
+		input []byte
+		gas   tosca.Gas
+		err   error
+	}{
+		"outOfGas": {
+			make([]byte, 96),
+			1,
+			ErrOutOfGas,
+		},
+		"invalidInput": {
+			make([]byte, 63),
+			20000,
+			ErrExecutionReverted,
+		},
+		"successful": {
+			make([]byte, 96),
+			20000,
+			nil,
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+			if test.err == nil {
+				state.EXPECT().SetStorage(tosca.Address{}, tosca.Key{}, tosca.Word{})
+			}
+
+			gas, err := stateSetStorage(state, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != 0 {
+				t.Errorf("unexpected gas, want %v, got %v", 0, gas)
+			}
+		})
+	}
+}
+
+func TestPrecompiled_StateIncNonce(t *testing.T) {
+	increment := byte(5)
+	tests := map[string]struct {
+		input  []byte
+		gas    tosca.Gas
+		sender tosca.Address
+		err    error
+	}{
+		"outOfGas": {
+			make([]byte, 64),
+			1,
+			tosca.Address{},
+			ErrOutOfGas,
+		},
+		"invalidInput": {
+			make([]byte, 63),
+			9000,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"sameAddress": {
+			make([]byte, 64),
+			9000,
+			tosca.Address{},
+			ErrExecutionReverted,
+		},
+		"invalidValue": {
+			make([]byte, 64),
+			9000,
+			tosca.Address{0x01},
+			ErrExecutionReverted,
+		},
+		"successful": {
+			append(append(make([]byte, 32), increment), make([]byte, 31)...),
+			9000,
+			tosca.Address{0x01},
+			nil,
+		},
+	}
+
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			state := tosca.NewMockWorldState(ctrl)
+			if test.err == nil {
+				state.EXPECT().GetNonce(tosca.Address{}).Return(uint64(5))
+				state.EXPECT().SetNonce(tosca.Address{}, uint64(5+increment))
+			}
+
+			gas, err := stateIncNonce(state, test.sender, test.input, test.gas)
+			if err != test.err {
+				t.Errorf("unexpected error, want %v, got %v", test.err, err)
+			}
+			if gas != 0 {
+				t.Errorf("unexpected gas, want %v, got %v", 0, gas)
+			}
+		})
+	}
+
+}
+
+func TestPrecompiled_HandleStatePrecompiled(t *testing.T) {
+	codeSize := tosca.Gas(42)
+	increment := byte(5)
+	tests := map[string]struct {
+		recipient          tosca.Address
+		inputPrefix        []byte
+		input              []byte
+		isStatePrecompiled bool
+		mockSetUp          func(*tosca.MockWorldState)
+	}{
+		"nonPrecompiled": {
+			tosca.Address{0x20},
+			[]byte{0x01, 0x02, 0x03, 0x04},
```

Comment:
Is there a test that checks that an insufficient long prefix is handled correctly?

Reply:
I added a test for invalid input.